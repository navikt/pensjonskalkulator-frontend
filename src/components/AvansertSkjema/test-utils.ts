export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
