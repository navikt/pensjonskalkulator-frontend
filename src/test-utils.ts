import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
