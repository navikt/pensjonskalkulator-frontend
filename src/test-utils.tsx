import React, { PropsWithChildren } from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import { PreloadedState, createListenerMiddleware } from '@reduxjs/toolkit'
import { render, RenderOptions } from '@testing-library/react'

import { getTranslation_test } from '@/utils/__tests__/test-translations'

import { createUttaksalderListener } from './state/listeners/uttaksalderListener'
import {
  setupStore,
  RootState,
  AppStore,
  AppStartListening,
} from './state/store'
import { getTranslation_nb } from './translations/nb'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
  hasRouter?: boolean
}

export const swallowErrors = (testFn: () => void) => {
  const error = console.error
  console.error = () => {}
  testFn()
  console.error = error
}

export const swallowErrorsAsync = async (testFn: () => Promise<void>) => {
  const cache = console.error
  console.error = () => {}
  await testFn()
  console.error = cache
}

function generateMockedTranslations() {
  const nbTranslations: Record<string, string> = getTranslation_nb()
  const testTranslations: Record<string, string> = getTranslation_test()
  const translationsInput = { ...nbTranslations, ...testTranslations }
  const translations: Record<string, string> = {}

  for (const key in translationsInput) {
    if (
      /<(?=.*? .*?\/ ?>|br|hr|input|!--|wbr)[a-z]+.*?>|<([a-z]+).*?<\/\1>|{[^}]+}/i.test(
        translationsInput[key]
      )
    ) {
      // for html or chunks keys: results in 'my_key' : 'my_key with some <html> or {chunk}'
      translations[key] = translationsInput[key]
    } else {
      // results in 'my_key' : 'my_key'
      translations[key] = key
    }
  }
  return translations
}

// Return an object with the store and all of RTL's query functions
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState, true),
    hasRouter = true,
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Provider store={store}>
        <IntlProvider locale="nb" messages={generateMockedTranslations()}>
          {hasRouter ? <MemoryRouter>{children}</MemoryRouter> : children}
        </IntlProvider>
      </Provider>
    )
  }

  const listenerMiddleware = createListenerMiddleware()
  createUttaksalderListener(
    listenerMiddleware.startListening as AppStartListening
  )

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'

export { renderWithProviders as render }
