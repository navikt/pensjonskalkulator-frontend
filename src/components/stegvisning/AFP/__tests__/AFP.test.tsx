import * as ReactRouterUtils from 'react-router'

import { describe, it, vi } from 'vitest'

import { AFP } from '..'
import { screen, render, waitFor, userEvent } from '@/test-utils'

describe('stegvisning - AFP', () => {
  const onCancelMock = vi.fn()
  const onPreviousMock = vi.fn()
  const onNextMock = vi.fn()

  it('kaller navigate når shouldRedirectTo er angitt', async () => {
    const navigateMock = vi.fn()
    vi.spyOn(ReactRouterUtils, 'useNavigate').mockImplementation(
      () => navigateMock
    )
    const randomPath = '/random-path'

    render(
      <AFP
        shouldRedirectTo={randomPath}
        afp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(navigateMock).toHaveBeenCalledWith(randomPath)
  })

  it('rendrer slik den skal når afp ikke er oppgitt', async () => {
    const user = userEvent.setup()
    const result = render(
      <AFP
        afp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
      'stegvisning.afp.title'
    )

    await user.click(screen.getByText('stegvisning.afp.readmore_privat_title'))
    await user.click(
      screen.getByText('stegvisning.afp.readmore_offentlig_title')
    )

    expect(result.asFragment()).toMatchSnapshot()
    expect(
      screen.getByRole('link', {
        name: 'AFP i privat sektor på afp.no application.global.external_link',
      })
    ).toHaveAttribute('href', 'https://www.afp.no')

    const radioButtons = await screen.findAllByRole('radio')
    await waitFor(() => {
      expect(radioButtons).toHaveLength(4)
      expect(radioButtons[0]).not.toBeChecked()
      expect(radioButtons[1]).not.toBeChecked()
      expect(radioButtons[2]).not.toBeChecked()
      expect(radioButtons[3]).not.toBeChecked()
      expect(result.asFragment()).toMatchSnapshot()
    })
  })

  it('rendrer slik den skal når afp er oppgitt', async () => {
    const result = render(
      <AFP
        afp="nei"
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
      'stegvisning.afp.title'
    )
    const radioButtons = screen.getAllByRole('radio')
    await waitFor(() => {
      expect(radioButtons).toHaveLength(4)
      expect(radioButtons[0]).not.toBeChecked()
      expect(radioButtons[1]).not.toBeChecked()
      expect(radioButtons[2]).toBeChecked()
      expect(radioButtons[3]).not.toBeChecked()
      expect(result.asFragment()).toMatchSnapshot()
    })
  })

  it('viser riktig infomeldinger når brukeren klikker på de ulike valgene', async () => {
    const user = userEvent.setup()
    render(
      <AFP
        afp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(screen.getByRole('heading', { level: 2 })).toHaveTextContent(
      'stegvisning.afp.title'
    )
    const radioButtons = screen.getAllByRole('radio')
    expect(
      screen.queryByText('stegvisning.afp.alert_vet_ikke')
    ).not.toBeInTheDocument()

    await user.click(radioButtons[0])

    expect(
      screen.queryByText('stegvisning.afp.alert_vet_ikke')
    ).not.toBeInTheDocument()

    await user.click(radioButtons[1])

    expect(
      screen.queryByText('stegvisning.afp.alert_vet_ikke')
    ).not.toBeInTheDocument()

    await user.click(radioButtons[2])

    expect(
      screen.queryByText('stegvisning.afp.alert_vet_ikke')
    ).not.toBeInTheDocument()

    await user.click(radioButtons[3])

    expect(
      screen.queryByText('stegvisning.afp.alert_vet_ikke')
    ).toBeInTheDocument()
  })

  it('validerer, viser feilmelding, fjerner feilmelding og kaller onNext når brukeren klikker på Neste', async () => {
    const user = userEvent.setup()
    render(
      <AFP
        afp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    const radioButtons = screen.getAllByRole('radio')

    await user.click(screen.getByText('stegvisning.neste'))

    waitFor(() => {
      expect(
        screen.getByText('stegvisning.afp.validation_error')
      ).toBeInTheDocument()
      expect(onNextMock).not.toHaveBeenCalled()
    })

    await user.click(radioButtons[0])

    expect(
      screen.queryByText('stegvisning.afp.validation_error')
    ).not.toBeInTheDocument()

    await user.click(screen.getByText('stegvisning.neste'))

    waitFor(() => {
      expect(onNextMock).toHaveBeenCalled()
    })
  })

  it('kaller onNext når brukeren klikker på Neste', async () => {
    const user = userEvent.setup()
    render(
      <AFP
        afp={null}
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    const radioButtons = screen.getAllByRole('radio')

    await user.click(radioButtons[0])
    await user.click(screen.getByText('stegvisning.neste'))

    waitFor(() => {
      expect(onNextMock).toHaveBeenCalled()
    })
  })

  it('kaller onPrevious når brukeren klikker på Tilbake', async () => {
    const user = userEvent.setup()
    render(
      <AFP
        afp="ja_privat"
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )

    await user.click(screen.getByText('stegvisning.tilbake'))

    expect(onPreviousMock).toHaveBeenCalled()
  })

  it('kaller onCancel når brukeren klikker på Avbryt', async () => {
    const user = userEvent.setup()
    render(
      <AFP
        afp="ja_privat"
        onCancel={onCancelMock}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )

    await user.click(screen.getByText('stegvisning.avbryt'))

    expect(screen.getByText('stegvisning.avbryt')).toBeInTheDocument()
    expect(onCancelMock).toHaveBeenCalled()
  })

  it('viser ikke avbryt knapp når onCancel ikke er definert', async () => {
    render(
      <AFP
        afp={null}
        onCancel={undefined}
        onPrevious={onPreviousMock}
        onNext={onNextMock}
      />
    )
    expect(screen.queryByText('stegvisning.avbryt')).not.toBeInTheDocument()
  })
})
