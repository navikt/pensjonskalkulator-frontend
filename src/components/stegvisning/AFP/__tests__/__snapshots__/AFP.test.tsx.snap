// Vitest Snapshot v1

exports[`stegvisning - AFP > rendrer slik den skal n책r afp er oppgitt 1`] = `
<DocumentFragment>
  <section
    aria-live="polite"
    class="_section_aded60 _section__largePadding_aded60 _section__marginBotton_aded60"
  >
    <form>
      <h2
        class="navds-heading navds-heading--medium navds-typo--spacing"
      >
        stegvisning.afp.title
      </h2>
      <p
        class="navds-ingress"
      >
        stegvisning.afp.ingress
      </p>
      <div
        class="navds-read-more navds-read-more--medium _readmorePrivat_10a2c6"
      >
        <button
          aria-expanded="false"
          class="navds-read-more__button navds-body-short"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            stegvisning.afp.readmore_privat_title
          </span>
        </button>
        <div
          aria-hidden="true"
          class="navds-read-more__content navds-read-more__content--closed navds-body-long navds-body-long--medium"
        >
          stegvisning.afp.readmore_list_title
          <ul
            class="_list_10a2c6"
          >
            <li>
              stegvisning.afp.readmore_privat_list_item1
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item2
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item3
            </li>
          </ul>
          Les om vilk책rene til 
          <a
            class="navds-link navds-link--action"
            href="stegvisning.afp.readmore_privat_url"
            target="_blank"
          >
            AFP i privat sektor p책 afp.no
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1.25rem"
              role="img"
              viewBox="0 0 24 24"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.75 4a.75.75 0 0 1 .75-.75H20a.748.748 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0V5.81l-8.72 8.72a.75.75 0 1 1-1.06-1.06l8.72-8.72H12.5a.75.75 0 0 1-.75-.75ZM5 8.75a.25.25 0 0 0-.25.25v10c0 .138.112.25.25.25h10a.25.25 0 0 0 .25-.25v-7a.75.75 0 0 1 1.5 0v7A1.75 1.75 0 0 1 15 20.75H5A1.75 1.75 0 0 1 3.25 19V9c0-.966.784-1.75 1.75-1.75h7a.75.75 0 0 1 0 1.5H5Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="navds-read-more navds-read-more--medium _readmoreOffentlig_10a2c6"
      >
        <button
          aria-expanded="false"
          class="navds-read-more__button navds-body-short"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            stegvisning.afp.readmore_offentlig_title
          </span>
        </button>
        <div
          aria-hidden="true"
          class="navds-read-more__content navds-read-more__content--closed navds-body-long navds-body-long--medium"
        >
          stegvisning.afp.readmore_list_title
          <ul
            class="_list_10a2c6"
          >
            <li>
              stegvisning.afp.readmore_offentlig_list_item1
            </li>
            <li>
              stegvisning.afp.readmore_offentlig_list_item2
            </li>
            <li>
              stegvisning.afp.readmore_offentlig_list_item3
            </li>
          </ul>
          stegvisning.afp.readmore_offentlig_ingress
        </div>
      </div>
      <fieldset
        aria-required="true"
        class="_radiogroup_10a2c6 navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
      >
        <legend
          class="navds-fieldset__legend navds-label"
        >
          stegvisning.afp.radio_label
        </legend>
        <div
          class="navds-radio-buttons"
        >
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-re"
              name="afp"
              type="radio"
              value="ja_offentlig"
            />
            <label
              class="navds-radio__label"
              for="radio-re"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_ja_offentlig
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-rf"
              name="afp"
              type="radio"
              value="ja_privat"
            />
            <label
              class="navds-radio__label"
              for="radio-rf"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_ja_privat
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              checked=""
              class="navds-radio__input"
              id="radio-rg"
              name="afp"
              type="radio"
              value="nei"
            />
            <label
              class="navds-radio__label"
              for="radio-rg"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_nei
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-rh"
              name="afp"
              type="radio"
              value="vet_ikke"
            />
            <label
              class="navds-radio__label"
              for="radio-rh"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_vet_ikke
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          class="navds-fieldset__error"
          id="fieldset-error-rd"
        />
      </fieldset>
      <button
        class="_button_10a2c6 navds-button navds-button--primary navds-button--medium"
        type="submit"
      >
        <span
          class="navds-label"
        >
          stegvisning.neste
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.tilbake
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--tertiary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.avbryt
        </span>
      </button>
       
    </form>
  </section>
</DocumentFragment>
`;

exports[`stegvisning - AFP > rendrer slik den skal n책r afp ikke er oppgitt 1`] = `
<DocumentFragment>
  <section
    aria-live="polite"
    class="_section_aded60 _section__largePadding_aded60 _section__marginBotton_aded60"
  >
    <form>
      <h2
        class="navds-heading navds-heading--medium navds-typo--spacing"
      >
        stegvisning.afp.title
      </h2>
      <p
        class="navds-ingress"
      >
        stegvisning.afp.ingress
      </p>
      <div
        class="navds-read-more navds-read-more--medium _readmorePrivat_10a2c6 navds-read-more--open"
      >
        <button
          aria-expanded="true"
          class="navds-read-more__button navds-body-short"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            stegvisning.afp.readmore_privat_title
          </span>
        </button>
        <div
          aria-hidden="false"
          class="navds-read-more__content navds-body-long navds-body-long--medium"
        >
          stegvisning.afp.readmore_list_title
          <ul
            class="_list_10a2c6"
          >
            <li>
              stegvisning.afp.readmore_privat_list_item1
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item2
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item3
            </li>
          </ul>
          Les om vilk책rene til 
          <a
            class="navds-link navds-link--action"
            href="stegvisning.afp.readmore_privat_url"
            target="_blank"
          >
            AFP i privat sektor p책 afp.no
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1.25rem"
              role="img"
              viewBox="0 0 24 24"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.75 4a.75.75 0 0 1 .75-.75H20a.748.748 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0V5.81l-8.72 8.72a.75.75 0 1 1-1.06-1.06l8.72-8.72H12.5a.75.75 0 0 1-.75-.75ZM5 8.75a.25.25 0 0 0-.25.25v10c0 .138.112.25.25.25h10a.25.25 0 0 0 .25-.25v-7a.75.75 0 0 1 1.5 0v7A1.75 1.75 0 0 1 15 20.75H5A1.75 1.75 0 0 1 3.25 19V9c0-.966.784-1.75 1.75-1.75h7a.75.75 0 0 1 0 1.5H5Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="navds-read-more navds-read-more--medium _readmoreOffentlig_10a2c6 navds-read-more--open"
      >
        <button
          aria-expanded="true"
          class="navds-read-more__button navds-body-short"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            stegvisning.afp.readmore_offentlig_title
          </span>
        </button>
        <div
          aria-hidden="false"
          class="navds-read-more__content navds-body-long navds-body-long--medium"
        >
          stegvisning.afp.readmore_list_title
          <ul
            class="_list_10a2c6"
          >
            <li>
              stegvisning.afp.readmore_offentlig_list_item1
            </li>
            <li>
              stegvisning.afp.readmore_offentlig_list_item2
            </li>
            <li>
              stegvisning.afp.readmore_offentlig_list_item3
            </li>
          </ul>
          stegvisning.afp.readmore_offentlig_ingress
        </div>
      </div>
      <fieldset
        aria-required="true"
        class="_radiogroup_10a2c6 navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
      >
        <legend
          class="navds-fieldset__legend navds-label"
        >
          stegvisning.afp.radio_label
        </legend>
        <div
          class="navds-radio-buttons"
        >
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r5"
              name="afp"
              type="radio"
              value="ja_offentlig"
            />
            <label
              class="navds-radio__label"
              for="radio-r5"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_ja_offentlig
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r6"
              name="afp"
              type="radio"
              value="ja_privat"
            />
            <label
              class="navds-radio__label"
              for="radio-r6"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_ja_privat
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r7"
              name="afp"
              type="radio"
              value="nei"
            />
            <label
              class="navds-radio__label"
              for="radio-r7"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_nei
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r8"
              name="afp"
              type="radio"
              value="vet_ikke"
            />
            <label
              class="navds-radio__label"
              for="radio-r8"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_vet_ikke
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          class="navds-fieldset__error"
          id="fieldset-error-r4"
        />
      </fieldset>
      <button
        class="_button_10a2c6 navds-button navds-button--primary navds-button--medium"
        type="submit"
      >
        <span
          class="navds-label"
        >
          stegvisning.neste
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.tilbake
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--tertiary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.avbryt
        </span>
      </button>
       
    </form>
  </section>
</DocumentFragment>
`;

exports[`stegvisning - AFP > rendrer slik den skal n책r afp ikke er oppgitt 2`] = `
<DocumentFragment>
  <section
    aria-live="polite"
    class="_section_aded60 _section__largePadding_aded60 _section__marginBotton_aded60"
  >
    <form>
      <h2
        class="navds-heading navds-heading--medium navds-typo--spacing"
      >
        stegvisning.afp.title
      </h2>
      <p
        class="navds-ingress"
      >
        stegvisning.afp.ingress
      </p>
      <div
        class="navds-read-more navds-read-more--medium _readmorePrivat_10a2c6 navds-read-more--open"
      >
        <button
          aria-expanded="true"
          class="navds-read-more__button navds-body-short"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            stegvisning.afp.readmore_privat_title
          </span>
        </button>
        <div
          aria-hidden="false"
          class="navds-read-more__content navds-body-long navds-body-long--medium"
        >
          stegvisning.afp.readmore_list_title
          <ul
            class="_list_10a2c6"
          >
            <li>
              stegvisning.afp.readmore_privat_list_item1
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item2
            </li>
            <li>
              stegvisning.afp.readmore_privat_list_item3
            </li>
          </ul>
          Les om vilk책rene til 
          <a
            class="navds-link navds-link--action"
            href="stegvisning.afp.readmore_privat_url"
            target="_blank"
          >
            AFP i privat sektor p책 afp.no
            <svg
              aria-hidden="true"
              fill="none"
              focusable="false"
              height="1.25rem"
              role="img"
              viewBox="0 0 24 24"
              width="1.25rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M11.75 4a.75.75 0 0 1 .75-.75H20a.748.748 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0V5.81l-8.72 8.72a.75.75 0 1 1-1.06-1.06l8.72-8.72H12.5a.75.75 0 0 1-.75-.75ZM5 8.75a.25.25 0 0 0-.25.25v10c0 .138.112.25.25.25h10a.25.25 0 0 0 .25-.25v-7a.75.75 0 0 1 1.5 0v7A1.75 1.75 0 0 1 15 20.75H5A1.75 1.75 0 0 1 3.25 19V9c0-.966.784-1.75 1.75-1.75h7a.75.75 0 0 1 0 1.5H5Z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="navds-read-more navds-read-more--medium _readmoreOffentlig_10a2c6 navds-read-more--open"
      >
        <button
          aria-expanded="true"
          class="navds-read-more__button navds-body-short"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="navds-read-more__expand-icon"
            fill="none"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.97 9.47a.75.75 0 0 1 1.06 0L12 14.44l4.97-4.97a.75.75 0 1 1 1.06 1.06l-5.5 5.5a.75.75 0 0 1-1.06 0l-5.5-5.5a.75.75 0 0 1 0-1.06Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
          <span>
            stegvisning.afp.readmore_offentlig_title
          </span>
        </button>
        <div
          aria-hidden="false"
          class="navds-read-more__content navds-body-long navds-body-long--medium"
        >
          stegvisning.afp.readmore_list_title
          <ul
            class="_list_10a2c6"
          >
            <li>
              stegvisning.afp.readmore_offentlig_list_item1
            </li>
            <li>
              stegvisning.afp.readmore_offentlig_list_item2
            </li>
            <li>
              stegvisning.afp.readmore_offentlig_list_item3
            </li>
          </ul>
          stegvisning.afp.readmore_offentlig_ingress
        </div>
      </div>
      <fieldset
        aria-required="true"
        class="_radiogroup_10a2c6 navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
      >
        <legend
          class="navds-fieldset__legend navds-label"
        >
          stegvisning.afp.radio_label
        </legend>
        <div
          class="navds-radio-buttons"
        >
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r5"
              name="afp"
              type="radio"
              value="ja_offentlig"
            />
            <label
              class="navds-radio__label"
              for="radio-r5"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_ja_offentlig
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r6"
              name="afp"
              type="radio"
              value="ja_privat"
            />
            <label
              class="navds-radio__label"
              for="radio-r6"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_ja_privat
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r7"
              name="afp"
              type="radio"
              value="nei"
            />
            <label
              class="navds-radio__label"
              for="radio-r7"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_nei
                </span>
              </span>
            </label>
          </div>
          <div
            class="navds-radio navds-radio--medium"
          >
            <input
              class="navds-radio__input"
              id="radio-r8"
              name="afp"
              type="radio"
              value="vet_ikke"
            />
            <label
              class="navds-radio__label"
              for="radio-r8"
            >
              <span
                class="navds-radio__content"
              >
                <span
                  class="navds-body-short navds-body-short--medium"
                >
                  stegvisning.afp.radio_vet_ikke
                </span>
              </span>
            </label>
          </div>
        </div>
        <div
          aria-live="polite"
          aria-relevant="additions removals"
          class="navds-fieldset__error"
          id="fieldset-error-r4"
        />
      </fieldset>
      <button
        class="_button_10a2c6 navds-button navds-button--primary navds-button--medium"
        type="submit"
      >
        <span
          class="navds-label"
        >
          stegvisning.neste
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.tilbake
        </span>
      </button>
      <button
        class="_button_10a2c6 navds-button navds-button--tertiary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          stegvisning.avbryt
        </span>
      </button>
       
    </form>
  </section>
</DocumentFragment>
`;
